WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -454,87 +455,102 @@ private extension CollectOrderPaymentUseCase {
                                                paymentAlerts: any CardReaderTransactionAlertsProviding<AlertPresenter.AlertDetails>,
                                                paymentGatewayAccount: PaymentGatewayAccount,
                                                onCompletion: @escaping (Result<CardPresentCapturedPaymentData, Error>) -> ()) {
-        alertsPresenter.present(
-            viewModel: paymentAlerts.error(error: error,
-                                           tryAgain: { [weak self] in
-                                               // Cancel current payment
-                                               self?.paymentOrchestrator.cancelPayment() { [weak self] result in
-                                                   guard let self = self else { return }
-
-                                                   switch result {
-                                                   case .success, .failure(CardReaderServiceError.paymentCancellation(.noActivePaymentIntent)):
-                                                       // Retry payment
-                                                       self.attemptPayment(alertProvider: paymentAlerts,
-                                                                           paymentGatewayAccount: paymentGatewayAccount,
-                                                                           onCompletion: onCompletion)
-                                                   case .failure(let cancelError):
-                                                       // Inform that payment can't be retried.
-                                                       self.alertsPresenter.present(
-                                                        viewModel: paymentAlerts.nonRetryableError(error: cancelError) {
-                                                            onCompletion(.failure(error))
-                                                        })
+        receiptStateForFailedPayment(error: error, order: order) { [weak self] receiptState in
```

Comment:
The review view shows a lot of changes, but I only wrapped it in `receiptStateForFailedPayment(error: error, order: order) { [weak self] receiptState in`